### ISSUE TITLE
CODING 2. Implement OLS via Gradient Descent

### ISSUE THREAD

**Issue by ninanikitina** [ninanikitina/quant-ml-sprint#6]
**Goal**
Show that you can solve the same optimization iteratively (this is important for scale, for streaming, and for “talk me through optimization”).
**Do this**

In src/models.py, add:

def ols_gd(
    X, y,
    lr=1e-2,
    max_iter=10_000,
    tol=1e-6,
    standardize=True
):
    """
 Linear regression via gradient descent on MSE.

    Steps:
      1. (optional) standardize columns of X to mean=0, std=1 for faster convergence
      2. initialize beta = 0
      3. repeat:
            grad = -(X^T (y - X beta)) / n
            beta <- beta - lr * grad
         until ||grad|| < tol or max_iter reached

    Returns:
        beta              : learned coefficients (de-standardized if needed)
        losses            : list of loss values per checkpoint
        grad_norm_history : list of gradient norms
    """
    ...

**Also:**
- [ ] Track loss = mean_squared_error(y, X @ beta).
- [ ] Stop when gradient norm is below tol.
- [ ] If you standardize X, remember to unscale beta before returning it.

In notebooks/01_bias_variance_clt.ipynb, create a small cell that:
- Generates synthetic X, y.
- Fits with ols_normal_eq and ols_gd.
- Plots loss vs iteration for GD.

Prints both betas side by side.

**Output**
Function ols_gd in src/[models.py](http://models.py/).
A section in the notebook called “Closed-form OLS vs Gradient Descent”.
A figure, e.g. gd_loss_curve, showing that loss decreases.

Why
 When you interview, you can now say:
- "Here’s the analytic solution."
- "Here’s the iterative solution."
- "Here’s when each is preferable."
